idasen/__init__.py

```diff
@@ -235,31 +247,54 @@ async def move_to_target(self, target: float):
                 f"{self.MIN_HEIGHT:.3f}"
             )
 
-        previous_height = await self.get_height()
-        will_move_up = target > previous_height
-        while True:
-            height = await self.get_height()
-            difference = target - height
-            self._logger.debug(f"{target=} {height=} {difference=}")
-            if (height < previous_height and will_move_up) or (
-                height > previous_height and not will_move_up
-            ):
-                self._logger.warning(
-                    "stopped moving because desk safety feature kicked in"
-                )
-                return
-            if abs(difference) < 0.005:  # tolerance of 0.005 meters
-                self._logger.info(f"reached target of {target:.3f}")
-                await self.stop()
-                return
-            elif difference > 0:
-                await self.move_up()
-            elif difference < 0:
-                await self.move_down()
-            previous_height = height
+        if self._moving:
+            self._logger.error("Already moving")
+            return
+        self._moving = True
+
+        async def do_move():
+            previous_height = await self.get_height()
+            will_move_up = target > previous_height
+            while True:
+                height = await self.get_height()
+                difference = target - height
+                self._logger.debug(f"{target=} {height=} {difference=}")
+                if (height < previous_height and will_move_up) or (
+                    height > previous_height and not will_move_up
+                ):
+                    self._logger.warning(
+                        "stopped moving because desk safety feature kicked in"
+                    )
+                    return
+                if not self._moving:
+                    return
+
+                if abs(difference) < 0.005:  # tolerance of 0.005 meters
+                    self._logger.info(f"reached target of {target:.3f}")
+                    await self._stop()
+                    self._moving = False
+                    return
+                elif difference > 0:
+                    await self.move_up()
+                elif difference < 0:
+                    await self.move_down()
+                previous_height = height
+
+        self._move_task = asyncio.create_task(do_move())
+        await self._move_task
```

Comment:
```suggestion
        await self._move_task
        self._moving = False
```

Should this unset `self._moving` when it is done?  Otherwise it only gets set `False` with `stop`.